### PR TITLE
Remove GET url for the tool to get lang.

### PR DESCRIPTION
**Pull request by vrezhMikTumo**
Tool used to determine language by GET url (?lang=<value>). PluginAPI returns lang set by editor which can be used. Remove languageFromURL function. Add initData object which keeps language init.

### CODE REVIEW

**Review by HaykAvagyan1 — CHANGES_REQUESTED** [review]
There is some minor changes to be done, also did you check if this works in the path? If not thoroughly test that it works with all languages and that the language codes are the same before merging (you might have let's say GE for German and in activity manager it might be DE or vice versa).

File: src/anim.js [hunk 1]
```diff
@@ -12,13 +12,14 @@ let maxFr = 0;
 let currentFrame = [];
 let frameStack = [];
 let showTrajectory = true;
-let language = languageFromURL(window.location.href);
 var content;
 
 window.onload = async function () {
-  let getData = await getText(language);
-  content = getData;
+  let lang= await pluginAPI.initialize()
```
[the remark this answers is not in the document]
Comment (vrezhMikTumo) [reply src/anim.js:18]: done

File: src/anim.js [hunk 2]
```diff
@@ -12,13 +12,14 @@ let maxFr = 0;
...
 
+  let getData =await getText(lang.language);
+  content = getData;
```
[the remark this answers is not in the document]
Comment (vrezhMikTumo) [reply src/anim.js:21]: done

File: src/pluginAPI.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
 const pluginAPI = {
   initialize: async () => {
     postMessage("init");
-    let value = false;
-
+    const initData = {language:undefined };
+    initData.value = false
```
[the remark this answers is not in the document]
Comment (vrezhMikTumo) [reply src/pluginAPI.js:5]: done

File: src/pluginAPI.js [hunk 4]
```diff
@@ -12,15 +12,16 @@ const pluginAPI = {
         const { data } = event.data;
 
         if (data && data.answers) {
-          value = data.answers[0];
+          initData.language = data.language;
+          initData.value = data.answers[0];
           resolve();
           return;
-        }
+        }initData.value = false;
```
[the remark this answers is not in the document]
Comment (vrezhMikTumo) [reply src/pluginAPI.js:19]: done

File: src/anim.js [hunk 5]
```diff
@@ -12,13 +12,13 @@ let maxFr = 0;
 let currentFrame = [];
 let frameStack = [];
 let showTrajectory = true;
-let language = languageFromURL(window.location.href);
-var content;
 
 window.onload = async function () {
+  // let lang= await pluginAPI.initialize()
+  const language = (await pluginAPI.initialize()).language
```
Comment (HaykAvagyan1) [inline src/anim.js:18]: I think there should be a semicolon here for consistent code style.

File: src/examiner.js [hunk 6]
```diff
@@ -1,9 +1,9 @@
 const showAnswer = async function () {
   let answer = await pluginAPI.initialize();
-
-  pluginAPI.examine(answer);
+  console.log(answer.answer_state)
+  pluginAPI.examine(answer.answer_state);
   pluginAPI.setHeight(100);
-  document.getElementById("answerBox").style.background = answer
+  document.getElementById("answerBox").style.background = answer.answer_state
```
Comment (HaykAvagyan1) [inline src/examiner.js:6]: Semicolon missing

File: src/examiner.js [hunk 7]
```diff
@@ -1,9 +1,9 @@
 const showAnswer = async function () {
   let answer = await pluginAPI.initialize();
-
-  pluginAPI.examine(answer);
+  console.log(answer.answer_state)
```
Comment (HaykAvagyan1) [inline src/examiner.js:3]: Remove log.

**Review by HaykAvagyan1 — CHANGES_REQUESTED** [review]
Looks good, some very small changes. Did you test in path? Because I think the language code in activity-manager for Armenian is `hy` instead of `am` (please double check by looking at the `language` variable in the `init-response` or ask engineers).

File: src/examiner.js (hunk 7 above)
Comment (vrezhMikTumo) [reply src/examiner.js:3]: done

File: src/examiner.js (hunk 6 above)
Comment (vrezhMikTumo) [reply src/examiner.js:6]: done

File: src/anim.js (hunk 5 above)
Comment (vrezhMikTumo) [reply src/anim.js]: done

File: src/anim.js [hunk 8]
```diff
@@ -12,13 +12,12 @@ let maxFr = 0;
 let currentFrame = [];
 let frameStack = [];
 let showTrajectory = true;
-let language = languageFromURL(window.location.href);
-var content;
+let content;
 
 window.onload = async function () {
+  const language = (await pluginAPI.initialize()).language;
```
Comment (HaykAvagyan1) [inline src/anim.js:18]: Can you make this `const language = (await pluginAPI.initialize()).language || "en";` so the default language is English if we try to use it outside of activity manager

**Review by HaykAvagyan1 — CHANGES_REQUESTED** [review]
Looks good, small thing that would be nice to have and then we can merge.

**Review by HaykAvagyan1 — APPROVED** [review]
Nice!

**Pull request merged**